snap/snapenv/snapenv.go

```diff
@@ -106,16 +110,60 @@ func userEnv(info *snap.Info, home string) map[string]string {
 	return result
 }
 
-// envMap creates a map from the given environment string list, e.g. the
-// list returned from os.Environ()
-func envMap(env []string) map[string]string {
+// Environment variables glibc strips out when running a setuid binary.
+// Taken from https://sourceware.org/git/?p=glibc.git;a=blob_plain;f=sysdeps/generic/unsecvars.h;hb=HEAD
+var unsafeEnv = map[string]bool{
+	"GCONV_PATH":       true,
+	"GETCONF_DIR":      true,
+	"GLIBC_TUNABLES":   true,
+	"HOSTALIASES":      true,
+	"LD_AUDIT":         true,
+	"LD_DEBUG":         true,
+	"LD_DEBUG_OUTPUT":  true,
+	"LD_DYNAMIC_WEAK":  true,
+	"LD_HWCAP_MASK":    true,
+	"LD_LIBRARY_PATH":  true,
+	"LD_ORIGIN_PATH":   true,
+	"LD_PRELOAD":       true,
+	"LD_PROFILE":       true,
+	"LD_SHOW_AUXV":     true,
+	"LD_USE_LOAD_BIAS": true,
+	"LOCALDOMAIN":      true,
+	"LOCPATH":          true,
+	"MALLOC_TRACE":     true,
+	"NIS_PATH":         true,
+	"NLSPATH":          true,
+	"RESOLV_HOST_CONF": true,
+	"RES_OPTIONS":      true,
+	"TMPDIR":           true,
+	"TZDIR":            true,
+}
+
+const PreservedUnsafePrefix = "SNAP_SAVED_"
+
+// envMap creates a map from the given environment string list,
+// e.g. the list returned from os.Environ(). If preserveUnsafeVars
+// rename variables that will be stripped out by the dynamic linker
+// executing the setuid snap-confine by prepending their names with
+// PreservedUnsafePrefix.
+func envMap(env []string, preserveUnsafeVars bool) map[string]string {
```

Comment:
nitpick - I'm not a fan of this boolean here. It makes it hard to reason about what the function is doing without looking at the signature. I.e.: what does `envMap(env, true)` mean? I would prefer a flag argument and a name, maybe something like `envMap(env, {0, preserveUnsafeVars])` ?

Reply:
I'm not 100% what you mean, but see what I pushed?

Reply:
Yeah, this is what I had in mind, thanks for this! Sorry that I was hard to parse.